pallets/ajuna-board/src/dot4gravity.rs

```diff
@@ -0,0 +1,216 @@
+// Ajuna Node
+// Copyright (C) 2022 BlogaTech AG
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use ajuna_common::{Finished, TurnBasedGame};
+use dot4gravity::{Coordinates, Game as Dot4Gravity, GameState, RandomBlocksGenerator, Side};
+use frame_support::pallet_prelude::*;
+
+#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebugNoBound, TypeInfo, MaxEncodedLen)]
+pub enum Turn {
+	DropBomb(Coordinates),
+	DropStone((Side, u8)),
+}
```

Comment:
`Turn` is defined here in `ajuna-board` pallet since it's specific to pallet integration (ie, `TurnBasedGame` trait).